simulation-construction-set/falling-brick/src/main/java/us/ihmc/fallingBrick/FallingBrickRobot.java

```diff
@@ -0,0 +1,204 @@
+   private Link base(String name, AppearanceDefinition appearance)
+   {
+      // creates a new link with the name specified in the parameter
+      Link ret = new Link(name);
+      // sets the mass of the link
+      ret.setMass(M1);
+      // sets the moment of inertia
+      ret.setMomentOfInertia(Ixx1, Iyy1, Izz1);
+      // sets the center of mass offset
+      ret.setComOffset(0.0, 0.0, 0.0);
+      // Use to visually represent links in the SCS 3D view
+      Graphics3DObject linkGraphics = new Graphics3DObject();
+      /*
+       * Translates from the current position by the specified distances. Graphic components added after
+       * translation will appear in the new location. The coordinate system for these translations is
+       * based on those that preceded it.
+       */
+      linkGraphics.translate(0.0, 0.0, -B1);
+
+      // add the pyramid cube 
+      linkGraphics.addPyramidCube(BASE_L, BASE_W, BASE_H, BASE_H, appearance);
+
+      // Attach the Graphics3DObject to its link
+      /*
+       * Associates our linkGraphics object with the ret object and in doing so, translates and rotates
+       * the graphic components to be in the same frame of reference as the ret.
+       */
+      ret.setLinkGraphics(linkGraphics);
+
+      return ret;
+   }
+
+   /**
+    * This method sets the initial positions, velocities, and accelerations of the brick
+    */
+   public void initRobot()
+   {
+      q_qlength = new YoDouble("q_qlength", registry);
+      theta_x = new YoDouble("theta_x", registry);
+
+      // sets the time
+      t.set(0.0);
+
+      // gets the position, velocity, and acceleration variables from the YoVariable registry
+      q_x = (YoDouble) this.getVariable("q_x");
+      q_y = (YoDouble) this.getVariable("q_y");
+      q_z = (YoDouble) this.getVariable("q_z");
+      qd_x = (YoDouble) this.getVariable("qd_x");
+      qd_y = (YoDouble) this.getVariable("qd_y");
+      qd_z = (YoDouble) this.getVariable("qd_z");
+      qdd_x = (YoDouble) this.getVariable("qdd_x");
+      qdd_y = (YoDouble) this.getVariable("qdd_y");
+      qdd_z = (YoDouble) this.getVariable("qdd_z");
+
+      q_qs = (YoDouble) this.getVariable("q_qs");
+      q_qx = (YoDouble) this.getVariable("q_qx");
+      q_qy = (YoDouble) this.getVariable("q_qy");
+      q_qz = (YoDouble) this.getVariable("q_qz");
+      qd_wx = (YoDouble) this.getVariable("qd_wx");
+      qd_wy = (YoDouble) this.getVariable("qd_wy");
+      qd_wz = (YoDouble) this.getVariable("qd_wz");
+      qdd_wx = (YoDouble) this.getVariable("qdd_wx");
+      qdd_wy = (YoDouble) this.getVariable("qdd_wy");
+      qdd_wz = (YoDouble) this.getVariable("qdd_wz");
+
+      // sets the initial positions, velocities, and accelerations of the falling brick
+      q_x.set(0.0);
+      q_y.set(0.0);
+      q_z.set(0.6);
+
+      q_qs.set(0.707);
+      q_qx.set(0.3);
+      q_qy.set(0.4);
+      q_qz.set(0.5);
+
+      qd_wx.set(0.0001);
+      qd_wy.set(1.0);
+      qd_wz.set(0.5001);
```

Comment:
Well I guess my comment above about these variables was premature, although these can be accessed via the API of the `floatingJoint` which would be cleaner. Alright so I guess line 35 declare the `floatingJoint` as a `FloatingJoint` so you can have access to its API, then here you can directly use it to initialize its initial configuration.